### PR TITLE
mock user agent

### PR DESCRIPTION
**Pull request by rllin-fathom**
some blacklist rules sites require a user agent to download (otherwise returns a 403)

tested on my own deployment.

- [x] I signed Google's [Contributor License Agreement](https://opensource.google.com/docs/cla/)
- [x] My code conforms to Google's [python style](https://google.github.io/styleguide/pyguide.html)

### CODE REVIEW

**Comment by blueandgold** [comment]
@rllin-fathom Are you able to update the test here.  It should be easier to apply the mock to the new code.

File: google/cloud/forseti/scanner/audit/blacklist_rules_engine.py [hunk 1]
```diff
@@ -151,7 +151,9 @@ def get_and_parse_blacklist(url):
             lists: first one is IP addresses,
             second one is network blocks
         """
-        data = urllib2.urlopen(url).read()
+        req = urllib2.build_opener()
+        req.addheaders = [('User-Agent', 'Forseti blacklist rules engine agent')]
```
[the remark this answers is not in the document]
Comment (rllin-fathom) [reply google/cloud/forseti/scanner/audit/blacklist_rules_engine.py]: i can't confirm other than empirically, but i know there are sites out there that purposefully block user agents of common python packages like urllib or requests

File: google/cloud/forseti/scanner/audit/blacklist_rules_engine.py [hunk 2]
```diff
@@ -151,7 +151,9 @@ def get_and_parse_blacklist(url):
             lists: first one is IP addresses,
             second one is network blocks
         """
-        data = urllib2.urlopen(url).read()
+        req = urllib2.build_opener()
```
[the remark this answers is not in the document]
Comment (rllin-fathom) [reply google/cloud/forseti/scanner/audit/blacklist_rules_engine.py]: good point!

**Comment by dekuhn** [comment]
FYI: We have announced a code freeze starting May 16, 2019, and ending May 30, 2019. During this time we won't be merging any new pull requests. For more details you can read the full announcement details [here](https://groups.google.com/a/forsetisecurity.org/forum/#!topic/announce/iIV-sr5zfds).

**Comment by kevensen** [comment]
@dekuhn @rllin-fathom Is this PR still relevant?  Can this be updated or closed?

**Comment by blueandgold** [comment]
This PR prevents 403 on reqesting the blacklist, which is nice to have.  The one thing that was blocking the merge of this PR is to fix a test that was broken.

File: google/cloud/forseti/scanner/audit/blacklist_rules_engine.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (natking1987) [reply google/cloud/forseti/scanner/audit/blacklist_rules_engine.py]: n

**Pull request closed without merging**